src/benchmarks/micro/libraries/System.Utf8String.Experimental/Perf.Utf8String.cs

```diff
@@ -0,0 +1,69 @@
+        }
+
+        [Benchmark]
+        [ArgumentsSource(nameof(NonAsciiData))]
+        public void ToUtf16(string expected)
+        {
+            Utf8Span span = new Utf8Span(new Utf8String(expected));
+            Memory<char> memory = new char[expected.Length];
+            Span<char> destination = memory.Span;
+            span.ToChars(destination);
+        }
+
+        [Benchmark]
+        [ArgumentsSource(nameof(NonAsciiData))]
+        public void IsAscii(string expected)
+        {
+            Utf8Span span = new Utf8Span(new Utf8String(expected));
+            span.IsAscii();
+        }
+
+        [Benchmark]
+        [ArgumentsSource(nameof(AsciiData))]
+        public void IsAscii_GetIndexOfFirstNonAsciiByte(string expected)
+        {
+            Utf8Span span = new Utf8Span(new Utf8String(expected));
+            span.IsAscii();
+        }
+
+        [Benchmark]
+        [ArgumentsSource(nameof(AsciiData))]
+        public void IsNormalized_GetIndexOfFirstNonAsciiChar(string expected)
+        {
+            expected.IsNormalized();
+            expected.IsNormalized();
+        }
+
+        [Benchmark]
+        [ArgumentsSource(nameof(AsciiData))]
+        public void IsNormalized_WidenAsciiToUtf16(string expected)
+        {
+            Utf8Span span = new Utf8Span(new Utf8String(expected));
```

Comment:
hhm, the return value is not used anywhere. Did you check what happens to the actual calls to `ToCharArray()`? Is your code even getting called?

Reply:
It is. I did explicitly check for that before running the benchmark :) Others may have the same question though, so I'll add some lines here to consume the result of `ToCharArray` in some way